src/main/java/com/denizenscript/depenizen/bukkit/commands/quests/QuestsCommand.java

```diff
@@ -0,0 +1,221 @@
+                    && arg.matchesPrimitive(ArgumentHelper.PrimitiveType.Integer)) {
+                scriptEntry.addObject("points", arg.asElement());
+            }
+            
+            else if (!scriptEntry.hasObject("state")
+                    && arg.matchesPrefix("state")
+                    && arg.matchesEnum(State.values())) {
+                scriptEntry.addObject("state", arg.asElement());
+            }
+            else {
+                arg.reportUnhandled();
+            }
+
+        }
+
+        if (!scriptEntry.hasObject("action")) {
+            throw new InvalidArgumentsException("Must specify a valid action!");
+        }
+        
+        scriptEntry.defaultObject("state", new ElementTag("TRUE"));
+    }
+
+    @Override
+    public void execute(ScriptEntry scriptEntry) {
+        Quests quests = (Quests) QuestsBridge.instance.plugin;
+        
+        BukkitScriptEntryData scriptEntryData = (BukkitScriptEntryData) scriptEntry.entryData;
+
+        // Get objects
+        ElementTag action = scriptEntry.getElement("action");
+        ElementTag questId = scriptEntry.getElement("quest-id");
+        ElementTag stageNum = scriptEntry.getElement("stage-no");
+        ElementTag points = scriptEntry.getElement("points");
+        ElementTag state = scriptEntry.getElement("state");
+
+        PlayerTag player = scriptEntryData.getPlayer();
+
+        // Report to dB
+        Debug.report(scriptEntry, getName(), action.debug() 
+                + (questId != null ? questId.debug() : "") 
+                + (stageNum != null ? stageNum.debug() : "") 
+                + (points != null ? points.debug() : "") 
+                + (state != null ? state.debug() : ""));
+
+        switch (Action.valueOf(action.asString().toUpperCase())) {
+
+            case ADD: {
+
+                if (questId != null && state != null) {
+                    for (Quest quest : quests.getQuests()) {
+                        if (quest.getId().equals(questId.asString()) && player != null) {
+                            quests.getQuester(player.getPlayerEntity().getUniqueId()).takeQuest(quest, state.asBoolean());
+                            break;
+                        }
+                    }
+                }
+                else if (points != null && points.asInt() > 0) {
+                    Quester quester = quests.getQuester(player.getPlayerEntity().getUniqueId());
+                    quester.setQuestPoints(quester.getQuestPoints() + points.asInt());
+                    reloadData(quester);
+                }
+                
+                break;
+
+            }
+            case REMOVE: {
+                
+                if (questId != null && state != null) {
+                    for (Quest quest : quests.getQuests()) {
+                        if (quest.getId().equals(questId.asString()) && player != null) {
+                            Player p = player.getPlayerEntity();

```

Comment:
Please use clearer variable names like `player` instead of `p`. Since `player` is already a defined variable, and you're just trying to get the bukkit player, consider simply using `player.getPlayerEntity()` in the getUniqueId and sendMessage calls (isOnline is available via PlayerTag already anyway).

Reply:
Since the `p` variable is referenced five times, I disagree on using `player.getEntity()` (if not for efficiency but for readability) but am all for more descriptive variable names. Will address.

Reply:
It would only need to be directly added to two locations, not five.

Reply:
Sorry, looks like I meant four:

```
Player **p** = player.getPlayerEntity();
Quester quester = quests.getQuester(**p**.getUniqueId());
quester.hardQuit(quest);
if (state.asBoolean() == true && **p**.isOnline()) {
    **p**.sendMessage(Lang.get(**p**, "questQuit").replace("<quest>", quest.getName()));
}
```

Note that I've since changed `p` to `thePlayer`.

Reply:
Two.
Two.
1, 2.
getUniqueId, and sendMessage. Two.

isOnline works on the original `player` variable.
The variable declaration would obviously not remain in the situation that the variable is removed.

Please do not mark resolved a comment that you didn't resolve.

Reply:
No, I meant that it's referenced four times. Once in #getQuester, once for #isOnline and twice (once for, once in) #sendMessage. Sorry for the confusion.

I'll go ahead and do as you ask, but by this logic we may as well call `quests.getQuester(player.getPlayerEntity().getUniqueId())` every time since `quests` is already defined. At this point I think it'd be best for you to make any desired changes post-merge.